File: Javascript/String/E-58. Length of Last Word.ts
function lengthOfLastWord(s: string): number {
  const trimS = s.trim();
  const n = trimS.length;
  for (let i = n - 1; i >= 0; i--) {
    if (trimS[i] === ' ') return n - i - 1;
  }
  return n;
}

const testData = [
  {
    s: 'Hello World',
    ans: 5,
  },
  {
    s: '   fly me   to   the moon  ',
    ans: 4,
  },
  { s: 'luffy is still joyboy', ans: 6 },
];

for (const data of testData) {
  console.log(data.ans === lengthOfLastWord(data.s));
}

File: Python/String/E-58. Length of Last Word.py
class Solution:
    @classmethod
    def lengthOfLastWord(self, s: str) -> int:
        trim_s = s.strip()
        n = len(trim_s)

        for i in range(n - 1, 0, -1):
            if trim_s[i] == " ":
                return n - i - 1

        return n


testData = [
    {
        "s": "Hello World",
        "ans": 5,
    },
    {
        "s": "   fly me   to   the moon  ",
        "ans": 4,
    },
    {"s": "luffy is still joyboy", "ans": 6},
]

for data in testData:
    print(data["ans"] == Solution.lengthOfLastWord(data["s"]))
